redbiom/admin.py

```diff
@@ -0,0 +1,324 @@
+
+    payload = "samples-represented-observations/%s" % '/'.join(samples)
+    post(context, 'SADD', payload)
+
+
+def load_sample_data(table, context):
+    """Load nonzero sample data.
+
+    Parameters
+    ----------
+    table : str, filepath
+        The filepath to a BIOM table to load.
+    context : str
+        The context to load into.
+
+    Raises
+    ------
+    ValueError
+        If the context to load into does not exist.
+        If any sample in the table already has been loaded.
+        If a samples metadata has not already been loaded.
+
+    Notes
+    -----
+    This method does not support non count data.
+
+    This load is formed to be performed in serial within a given context. It
+    is safe to issue load commands in parallel as a lock within a context is
+    obtained prior to performing the load.
+
+    The observation IDs are remapped into an integer space to reduce memory
+    consumption as sOTUs are large. The index is maintained in Redis.
+
+    The indexing scheme requires a central authority for obtaining a unique
+    and stable index value. As such, this method obtains a lock for performing
+    an update. The index is updated with any new observations seen on each
+    load and is stored under the "__observation_index" key. It is stored as
+    a JSON object mapping the original observation ID to a stable and unique
+    integer; stability and uniqueness is not assured across distinct redis
+    databases.
+
+    The data are stored per sample with keys of the form "data:<sample_id>".
+    The string stored is tab delimited, where the even indices (i.e .0, 2, 4,
+    etc) correspond to the unique index value for an observation ID, and the
+    odd indices correspond to the counts associated with the sample/observation
+    combination.
+
+    Redis command summary
+    ---------------------
+    SMEMBERS <context>:samples-represented-observations
+    SETNX <context>:__load_table_lock 1
+    GET <context>:__observation_index
+    SET <context>:data:<sample_id> <packed-nz-representation>
+    SET <context>:__observation_index <revised-observation-mappings>
+    DEL <context>:__load_table_lock
+    SADD <context>:samples-represented-data <sample_id> ... <sample_id>
+    """
+    import biom
+    import time
+    import json
+    import redbiom
+    import redbiom.requests
+    import redbiom.util
+
+    config = redbiom.get_config()
+    post = redbiom.requests.make_post(config)
+    get = redbiom.requests.make_get(config)
+
+    redbiom.requests.valid(context, get)
+
+    tab = biom.load_table(table)
+    obs = tab.ids(axis='observation')
+    samples = tab.ids()
+
+    if not redbiom.util.has_sample_metadata(samples):
+        raise ValueError("Sample metadata must be loaded first.")
+
+    represented = get(context, 'SMEMBERS', 'samples-represented-data')
+    if set(samples).intersection(set(represented)):
+        raise ValueError("At least one sample to load already exists")
+
+    acquired = False
```

Comment:
Can you create an issue? Decomposing it was low priority since `load_sample_data` is the only place the lock is actually needed. The lock in `load_metadata` was a dirty component of supporting duplicate sample IDs before we decided to just represent tags (prep IDs).